src/Directory.Build.props

```diff
@@ -35,8 +35,13 @@
     <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)CodeAnalysis.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
 
-  <!-- Only compile with IsAotCompatible on .NET 6.0 and later. Our Device Tests fail for net7.0-android with this enabled as well -->
-  <PropertyGroup Condition="'$(TargetFramework)' != 'net7.0-android' AND $([System.Text.RegularExpressions.Regex]::IsMatch('$(TargetFramework)', '^(?!net5\.|net4\d{2}|netstandard)net\d+'))">
+  <!-- Only compile with IsAotCompatible on .NET 8.0 and later -->
+  <PropertyGroup Condition="$([System.Text.RegularExpressions.Regex]::IsMatch('$(TargetFramework)', 'net8\.|net9\.|net\d{2,}\.'))">
```

Comment:
net9 and netXY? Aren't we getting ahead of ourselves? 😆

Reply:
Yeah probably... I was trying to write it so it wouldn't need to be changed next year when net9.0 is released... and then I added another 30 years or so runway after that (vast overkill) 😛 In reality, we're probably going to have to change it in a week or two when we discover it doesn't work with net8.0-android or something.